src/lazy/binary/raw/v1_1/type_descriptor.rs

```diff
@@ -14,6 +14,20 @@ pub struct Opcode {
 /// A statically defined array of TypeDescriptor that allows a binary reader to map a given
 /// byte (`u8`) to a `TypeDescriptor` without having to perform any masking or bitshift operations.
 pub(crate) static ION_1_1_OPCODES: &[Opcode; 256] = &init_opcode_cache();
+pub(crate) static ION_1_1_TYPED_NULL_TYPES: &[IonType; 12] = &[
+    IonType::Bool,
+    IonType::Int,
+    IonType::Float,
+    IonType::Decimal,
+    IonType::Timestamp,
+    IonType::String,
+    IonType::Symbol,
+    IonType::Blob,
+    IonType::Clob,
+    IonType::List,
+    IonType::SExp,
+    IonType::Struct,
+];
```

Comment:
Do you think there may be any benefit to using a `match` instead of a lookup table? E.g.:
```rust
#[inline]
fn get_type_of_null(null_body: u8) -> IonType {
    match null_body {
        0x0 => IonType::Bool,
        0x1 => IonType::Int,
        // ...
        0xB => IonType::Struct,
        _ => unreachable!(),
    }
}
```
If it isn't obvious which one is better and/or there is no consensus in the Rust community, then we should just create an issue to revisit `match` vs lookup tables at some later time. No need to go deep into a rabbit hole on this now.

Reply:
Wow, that was a very thorough answer for my idle curiosity.

> If the values for null_body were the same as the values used in IonType, then rust is smart enough to just return null_body and not lookup, or cmp anything. This would be awesome to make possible, but the binary encoded values would need to match the definition order of IonType (or we'd have to set the values).

Nice!

However, we already expect typed nulls to be a relatively cold path, so I guess it's probably not worth putting a lot of effort into optimizing it.